### PR TITLE
Add IO.gets

### PR DESCRIPTION
**Pull request by bmulvihill**
Adds gets to IO Module

### CODE REVIEW

File: spec/interpreter/native_lib/io_spec.cr [hunk 1]
```diff
@@ -55,4 +55,31 @@ describe "NativeLib - IO Methods" do
...
     end
   end
+
+  describe "#gets" do
+    it "no input" do
+      itr = interpret_with_mocked_input %q(
+        IO.gets
+      ), nil
+
+      itr.stack.pop.to_s.should eq("nil")
```
Comment (faultyserver) [inline spec/interpreter/native_lib/io_spec.cr]: This should probably be replaced with `itr.stack.pop.should eq(val(nil))`. The `to_s` for `TNil#to_s` might change to return a blank string, so I would like to keep things from depending on it. (in fact, the `to_s` on all of the Value classes might disappear entirely before too long. I added them long ago before native methods were really a thing.).

File: spec/interpreter/native_lib/io_spec.cr [hunk 2]
```diff
@@ -55,4 +55,31 @@ describe "NativeLib - IO Methods" do
...
+    end
+
+    it "a string from the IO" do
+      itr = interpret_with_mocked_input %q(
+        IO.gets               
+      ), "hello, world"
+
+      itr.stack.pop.to_s.should eq("hello, world")
```
Comment (faultyserver) [inline spec/interpreter/native_lib/io_spec.cr]: Same here and in the next test. Prefer `its.stack.pop.should eq(val("hello, world"))` :)

**Comment by faultyserver** [comment]
Looks great! Just have the two comments about the specs, then I'll go ahead and merge.

Thanks for the contribution! :)

**Comment by bmulvihill** [comment]
Updated

**Comment by faultyserver** [comment]
Awesome! Merging now.

Thanks again :)

**Pull request merged**